app/assets/javascripts/3dbio_viewer/src/webapp/components/molecular-structure/usePluginRef.ts

```diff
@@ -47,25 +56,58 @@ export function usePluginRef(options: Options) {
         setPdbePlugin,
     } = options;
 
+    // Set chain through molstar
+    const setChain = React.useCallback(
+        (chainId: string) => {
+            if (newSelection.ligandId !== undefined) {
+                console.debug("In ligand view. Not changing chain", newSelection.ligandId);
+                return;
+            }
+
+            if (newSelection.chainId === chainId) {
+                console.debug("Chain already set", chainId);
+                return;
+            }
+
+            console.debug("Set chain through molstar", chainId);
+            setSelection(setSelectionChain(newSelection, chainId));
+        },
+        [setSelection, newSelection]
+    );
+
+    const getLigandViewState = React.useMemo(() => () => newSelection.ligandId !== undefined, [
+        newSelection.ligandId,
+    ]);
```

Comment:
Unless there is something special, `React.useMemo(() => () =>` -> `React.useCallback(() => `

Reply:
It's the function sent to sequence component on pdbe-molstar. Will it work with the useCallback anyways? I think I tried with useCallback before and it didn't work, but now I don't remember quite well. Tomorrow I will test it 👌🏼

Reply:
It should work, both codes are conceptually equivalent.